src/raw_stream.rs

```diff
@@ -493,6 +520,96 @@ impl RawDevice {
             .map_err(nix_err)
     }
 
+    /// Update the auto repeat delays
+    #[inline]
+    pub fn update_auto_repeat(&self, repeat: &AutoRepeat) -> io::Result<()> {
+        let repeat = [repeat.delay, repeat.period];
+
+        unsafe { sys::eviocsrep(self.as_raw_fd(), &repeat) }
+            .map(|_| ())
+            .map_err(nix_err)
+    }
+
+    /// Retrieve the scancode for a keycode, if any
+    pub fn get_scancode_by_keycode(&self, keycode: u32) -> io::Result<Vec<u8>> {
+        let mut keymap = libc::input_keymap_entry {
+            flags: 0,
```

Comment:
I think using `..` for len, index, and scancode might work here to indicate we don't explicitly care about setting them.

Reply:
As far as I know we can't leave out structure fields in a structure literal. I've tried and it does not compile.

Reply:
I think it's `..Default::default()` I'm thinking of.

Reply:
Right, I did not know that. :smiley: 

However it does not work for input_keymap_entry:
```
error[E0277]: the trait bound `input_keymap_entry: Default` is not satisfied
   --> src/raw_stream.rs:545:15
    |
545 |             ..Default::default()
    |               ^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `input_keymap_entry`
    |
    = note: required by `std::default::Default::default`
```
I'm not sure it's worth creating PR on the libc crate to add `#[derive(Default)]`.

Reply:
Ah well, it was worth a shot. Yeah, I thought it would invoke Default piecemeal on each element but it looks like that's not the case.